src/lib/routers/history.ts

```diff
@@ -0,0 +1,200 @@
+import qs from 'qs';
+import { RouteState } from '../../types';
+
+type BrowserHistoryProps = {
+  windowTitle?: (routeState: RouteState) => string;
+  writeDelay: number;
+  createURL({
```

Comment:
We could extra the signature of the function outside of the type definition. That would be easier to follow.